x-pack/plugins/data_visualizer/public/application/index_data_visualizer/utils/saved_search_utils.test.ts

```diff
@@ -0,0 +1,313 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  getQueryFromSavedSearch,
+  createMergedEsQuery,
+  getEsQueryFromSavedSearch,
+} from './saved_search_utils';
+import type { SavedSearchSavedObject } from '../../../../common';
+import type { SavedSearch } from '../../../../../../../src/plugins/discover/public';
+import type { Filter, FilterStateStore } from '@kbn/es-query';
+import { stubbedSavedObjectIndexPattern } from '../../../../../../../src/plugins/data/common/data_views/data_view.stub';
+import { IndexPattern } from '../../../../../../../src/plugins/data/common';
+import { fieldFormatsMock } from '../../../../../../../src/plugins/field_formats/common/mocks';
+import { uiSettingsServiceMock } from 'src/core/public/mocks';
+
+// helper function to create data views
```

Comment:
Nit: It should probably still say index patterns instead of data views.

Reply:
Updated here https://github.com/elastic/kibana/pull/112870/commits/3aba6a59466d3b92ecfa7a9b7be5cea1cc1c3891